command/agent/command.go

```diff
@@ -472,6 +473,7 @@ func (c *Command) setupAgent(config *Config, logOutput io.Writer, logWriter *log
 		return err
 	}
 	c.agent = agent
+	agent.command = c
```

Comment:
This feels a little dirty to expose the whole command. How about `type reloadFn func() err` and pass that into `Create()`?